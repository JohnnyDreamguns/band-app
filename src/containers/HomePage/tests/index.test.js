import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { HomePage } from '../index';

const renderer = new ShallowRenderer();

const props = {
  page: 2,
  numOfPages: 5,
  loading: false,
  bands: [
    {
      id: 1,
      name: 'Band Name',
      bio: 'Band description'
    }
  ],
  loadBands: jest.fn(),
  setPreviousPage: jest.fn(),
  setNextPage: jest.fn(),
  clearHomepage: jest.fn()
};

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<HomePage {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
