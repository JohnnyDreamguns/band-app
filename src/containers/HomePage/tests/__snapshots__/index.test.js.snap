// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <p>
    <ForwardRef
      to="/band/1"
    >
      Band Name
    </ForwardRef>
  </p>
  <button
    disabled={false}
  >
    Previous Page
  </button>
  <button
    disabled={false}
  >
    Next Page
  </button>
</div>
`;
